src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java

```diff
@@ -135,41 +135,49 @@
 	private static final String FIELD_SCORE = "_score";
 
 	private Client client;
-	private ElasticsearchConverter elasticsearchConverter;
+	private ElasticsearchEntityConverter elasticsearchEntityConverter;
 	private ResultsMapper resultsMapper;
 	private String searchTimeout;
 
 	public ElasticsearchTemplate(Client client) {
-		this(client, new MappingElasticsearchConverter(new SimpleElasticsearchMappingContext()));
+		MappingElasticsearchConverter mappingElasticsearchConverter = createElasticsearchConverter();
+		initialize(client, mappingElasticsearchConverter,
+				new DefaultResultMapper(mappingElasticsearchConverter.getMappingContext()));
 	}
 
-	public ElasticsearchTemplate(Client client, EntityMapper entityMapper) {
-		this(client, new MappingElasticsearchConverter(new SimpleElasticsearchMappingContext()), entityMapper);
-	}
-
-	public ElasticsearchTemplate(Client client, ElasticsearchConverter elasticsearchConverter,
+	public ElasticsearchTemplate(Client client, ElasticsearchEntityConverter elasticsearchEntityConverter,
 			EntityMapper entityMapper) {
-		this(client, elasticsearchConverter,
-				new DefaultResultMapper(elasticsearchConverter.getMappingContext(), entityMapper));
+		initialize(client, elasticsearchEntityConverter,
+				new DefaultResultMapper(elasticsearchEntityConverter.getMappingContext(), entityMapper));
 	}
 
 	public ElasticsearchTemplate(Client client, ResultsMapper resultsMapper) {
```

Comment:
Let's also get rid of `ResultsMapper`. That is something to be moved into MappingElasticsearchConverter.

Reply:
next ticket I'll do